src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -239,34 +257,63 @@ bool emberAfGeneralCommissioningClusterCommissioningCompleteCallback(
         }
         else
         {
-            if (failSafe.NocCommandHasBeenInvoked())
+            CHIP_ERROR err;
+
+#if CHIP_CONFIG_TC_REQUIRED
+
+            uint16_t termsAndConditionsAcceptedAcknowledgements;
+            bool hasRequiredTermAccepted;
+            bool hasRequiredTermVersionAccepted;
+
+            err = enhancedSetupFlowProvider->GetTermsAndConditionsAcceptedAcknowledgements(
+                termsAndConditionsAcceptedAcknowledgements);
+            CheckSuccess(err, Failure);
+
+            err = enhancedSetupFlowProvider->HasTermsAndConditionsRequiredAcknowledgementsBeenAccepted(hasRequiredTermAccepted);
+            CheckSuccess(err, Failure);
+
+            err = enhancedSetupFlowProvider->HasTermsAndConditionsRequiredAcknowledgementsVersionBeenAccepted(
+                hasRequiredTermVersionAccepted);
+            CheckSuccess(err, Failure);
+
+            if (!hasRequiredTermAccepted)
             {
-                CHIP_ERROR err = fabricTable.CommitPendingFabricData();
-                if (err != CHIP_NO_ERROR)
-                {
-                    // No need to revert on error: CommitPendingFabricData always reverts if not fully successful.
-                    ChipLogError(FailSafe, "GeneralCommissioning: Failed to commit pending fabric data: %" CHIP_ERROR_FORMAT,
-                                 err.Format());
-                }
-                else
+                ChipLogProgress(AppServer, "Required terms and conditions have not been accepted");
+                Breadcrumb::Set(commandPath.mEndpointId, 0);
```

Comment:
- Don't set breadcrumb to zero: only ending failsafe should do that, since CommissioningComplete could be retried after attempting to resolve the issue.